cleanup

friej715 opened the pull request:
This PR looks huge, but 50% of the changes are just random linter things in files that haven't otherwise been touched for a while. Basically everything that matters is, again, in Generate.js.

Mostly what this PR does is change what format the Generate page is expecting for the filter data. 
- Before, the format was like this:
```
"filterA": [
    {
      "name": "gender",
      "choices": [
        "Female"
      ],
      "logic": "or"
    }
  ],
```
- Now, the format is like this:
```
"filterA":{
     "demographics":[
          {"operator":"or","values":[17]}
     ]
}
```

This mirrors what we're sending the server when we ask for two demographics to compare. Here, that format is decoded in the `decodeFilterRequestItem` function, after which it's sent to the usual sentence generator.

There are also some small bugfixes, and some attempts to better adhere to what Flow is expecting.

friej715 committed:
cleanup

friej715 committed:
console

chanderson0 commented on floodwatch/src/js/components/Generate.js:
```diff
@@ -1,54 +1,95 @@
-import type {VisibilityMap} from './Compare'
 
+type StateType = {
+  filterA: any,
```
`FilterA` and `FilterB` are known types - they're `FilterRequestItem`s. Ditto on `dataA` and `dataB` being known.

chanderson0 commented on floodwatch/src/js/components/Generate.js:
```diff
@@ -1,54 +1,95 @@
-    const lSentence = createSentence(curData.filterA);
-    const rSentence = createSentence(curData.filterB);
+  return {
+    dataA: (curData.dataA) ? curData.dataA : {},
```
You can use `curData.dataA || {}` for conciseness.

chanderson0 commented on floodwatch/src/js/components/Generate.js:
```diff
@@ -67,3 +108,68 @@ export class Generate extends Component {
+
+function decodeFilterRequestItem(filter: FilterRequestItem): Array<Filter> {
+  const keys = _.keys(filter)
+  const isPersonal = (_.indexOf(keys, 'personal') > -1);
```
Maybe `keys.includes('personal')`?

chanderson0 commented on floodwatch/src/js/components/Generate.js:
```diff
@@ -67,3 +108,68 @@ export class Generate extends Component {
+          newObj.logic = o.operator;
+          newObj.choices = [];
+          o.values.forEach((v) => {
+            let choice = _.find(DemographicKeys.demographic_keys, (dk) => {
```
There's nothing wrong with the way this is written, but there are a couple of more concise ways to write this, just FYI:

```js
let choice = _.find(DemographicKeys.demographic_keys, dk => dk.id == v)
```

or, using some lodash magic:

```js
let choice = _.find(DemographicKeys.demographic_keys, { id: v })
```

chanderson0 commented on floodwatch/src/js/components/Generate.js:
```diff
@@ -67,3 +108,68 @@ export class Generate extends Component {
+    }]
+  }
+
+  let optionsArr = []
```
Another meta comment that doesn't need to be addressed now but for the future: you should prefer to use `const` over `let` wherever possible. By using `let` variables only for times you need to change the assignment of the variable, it's clearer which variables are intended to be changed vs not.

chanderson0 commented on floodwatch/src/js/components/comparisontools.js:
```diff
@@ -1,6 +1,6 @@
 //@flow
 
-const unknownId = "16"
+const unknownId = '16'
```
Didn't catch this in the last one, but global constants should be `CAPS_CASE`

chanderson0 commented on floodwatch/src/js/components/comparisontools.js:
```diff
@@ -9,45 +9,51 @@ import type {Preset, Filter, FilterLogic} from './filtertypes'
 import type {VisibilityMap} from './Compare'
 
 export function getVisibilityMap(lD, rD): VisibilityMap {
```
This function is actually a great example of `let` and `const` - `visibilityMap` can be `const` because it'll never change, whereas `comparisonData` has to get assigned.

chanderson0 commented on floodwatch/src/js/components/Generate.js:
```diff
@@ -1,54 +1,95 @@
+    filterA: (curData.filterA) ? curData.filterA : {},
+    filterB: (curData.filterB) ? curData.filterB : {},
+    curTopic: (curData.curTopic) ? curData.curTopic : null,
+    visibilityMap: visibilityMap,
```
ES2016 feature: you don't have to repeat same-named keys and variables, so this and the next few can just be:

```js
{
  ...
  lSentence,
  rSentence,
  sentence
}
```

friej715 replied:
Whoa, that rules! Super good to know.

chanderson0 commented on floodwatch/src/js/components/Generate.js:
```diff
@@ -67,3 +108,68 @@ export class Generate extends Component {
+      })
+    } 
+    else if (k == 'demographics') {
+      if (filter[k].length > 0) {
```
This `if` is redundant... a for loop would just skip, so you don't need it.

chanderson0 reviewed (changes requested):
A few cleanup things, but in general this looks good. I haven't tried to run it yet, but we should be good to go shortly. 

Also, for next time: if you run `eslint --fix` and change a lot of unrelated files, can you just commit that directly without including it in the PR? I think we can agree with the eslint changes and it'd make this all a bit clearer.

friej715 committed:
cleaning up style stuff, fixing a sentence bug

friej715 commented:
Great. I think this last push addresses all these comments. Endlessly delighted by how much lodash can shorten code.

Last push also includes a bugfix--the comparison sentence was always "All Floodwatch users vs All Floodwatch users," for some reason.

friej715 committed:
console lol
